src/network/networkd-dhcp4.c

```diff
@@ -583,6 +586,47 @@ static int dhcp4_set_hostname(Link *link) {
         return sd_dhcp_client_set_hostname(link->dhcp_client, hn);
 }
 
+static bool check_promote_secondaries(const char *ifname) {
+        char *promote_secondaries_sysctl = NULL, *promote_secondaries_path = NULL;
+
+        promote_secondaries_path = strjoina("net/ipv4/conf/", ifname, "/promote_secondaries");
+        if (sysctl_read(promote_secondaries_path, &promote_secondaries_sysctl) >= 0) {
+                truncate_nl(promote_secondaries_sysctl);
+                return parse_boolean(promote_secondaries_sysctl);
```

Comment:
parse_boolean returns an int, not a bool, and that for a reason. It returns < 0 on parse failures and such.

While it is OK to ignore errors in this case, I figure this should be done explicitly, i.e. by changing this to ` return parse_boolean(…) > 0` or so, acknowledging that this is case, and explicitly making a choice whether parse erros are treated like "true" or like "false"...

BTW, if you want to gain extra points: instead of silently ignoring the sysctl_read() or parse_boolean() errors, consider adding at least a `log_debug_errno(r, 'something happened: %m')` for them, to keep things debuggable

Reply:
Thank you, that is a very good point. I updated the code to show a warning if the sysctl value cannot be parsed. As the kernel should set this to something sensible I think a warning is justified here. However happy to switch to debug if you think this is more appropriate.